resources/estimate_copy_number.R

```diff
@@ -0,0 +1,150 @@
+                    type = "calls",
+                    file = paste("%s.summarised_calls_binsize_",binsize,".tsv", sep = ""))
+
+## Raw calls - one file that contains *all* samples together 
+### MAY NOT NEED THIS FUNCTION
+QDNAseq::exportBins(copyNumbersCalled, 
+                    format = "tsv", 
+                    type = "calls",
+                    file = paste(filename, ".all_calls_binsize_",
+                                 binsize, "_merged.raw_calls_all_depths.tsv",
+                                 sep = ""))
+
+################## ACE FUNCTIONS #############################
+
+generate_sample_model <- function(object, QDNAseqobjectsample, filename){
+  # Perform model fitting on a single sample
+  model1 <- singlemodel(object, QDNAseqobjectsample = QDNAseqobjectsample)
+  # Select best fit by examining error lists (cellularity where relative error is lowest)
+  bestfit1 <- model1$minima[tail(which(model1$rerror==min(model1$rerror)), 1)]
+  besterror1 <- min(model1$rerror)
+  lastfit1 <- tail(model1$minima, 1)
+  lasterror1 <- tail(model1$rerror, 1)
+  error_plot <- print(model1$errorplot + ggtitle(paste0("sample ", filename, " - errorlist")) +
+                        theme(plot.title = element_text(hjust = 0.5)))
+  #  Use best fit and best error to select best model for absolute CN plot
+  png(paste0(filename,"_errorplot.png"))
+  ggsave(paste0(filename, "_errorplot.png"), error_plot, limitsize = FALSE)
+  dev.off()
+  
+  return(model1)
+}
+
+generate_absolute_plot <- function(object, QDNAseqobjectsample, bestfit1, 
+                                   besterror1, model1, filename) {
+  absolute_plot <- singleplot(object, QDNAseqobjectsample = QDNAseqobjectsample, cellularity = bestfit1,
+                              error = besterror1, standard = model1$standard,
+                              title = paste0(filename, "- binsize", binsize, "kbp_- 2N fit 1"))
+  # Create absolute CN plot
+  png(paste0(filename,"_plot.png"))
+  ggsave(paste0(filename, "_absoluteCN_plot.png"), absolute_plot, limitsize = FALSE)
+  dev.off()
+}
+
+generate_CN_dfs <- function(object, QDNAseqobjectsample, filename) {
+  template <- objectsampletotemplate(object, index = QDNAseqobjectsample)
+  write.table(template, file = paste0(filename,"_absolute_copy_number.tsv"),
+              sep = "\t")
+  adjusted_segments <- getadjustedsegments(template, log = TRUE)
+  write.table(adjusted_segments, file = paste0(filename,"_adjusted_segments.tsv"),
+              sep =	"\t")
+}
+
+run_ace_pipeline <- function(object, QDNAseqobjectsample, filename){
+  model1 <- generate_sample_model(object, QDNAseqobjectsample, filename)
+  bestfit1 <- model1$minima[tail(which(model1$rerror==min(model1$rerror)), 1)]
+  besterror1 <- min(model1$rerror)
+  generate_absolute_plot(object, QDNAseqobjectsample, bestfit1, besterror1, model1, filename)
+  generate_CN_dfs(object, QDNAseqobjectsample, filename)
+}
+
+run_ace_pipeline(copyNumbersCalled, QDNAseqobjectsample, filename)
```

Comment:
Please generate a few example outputs and share them to be reviewed together with the source code.